encoding/thrift/thriftrw-plugin-yarpc/main.go

```diff
@@ -0,0 +1,258 @@
+			"<.ThriftName>": <$thrift>.HandlerFunc(s.h.<.Name>),
+	<end>}
+}
+
+type handler struct{ impl Interface }
+
+<$service := .Service>
+<range .Service.Functions>
+
+<$servicePackage := import $service.Package>
+<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+
+func (h handler) <.Name>(
+	ctx <$context>.Context,
+	reqMeta <$yarpc>.ReqMeta,
+	body <$wire>.Value,
+) (<$thrift>.Response, error) {
+	var args <$servicePackage>.<.Name>Args
+	if err := args.FromWire(body); err != nil {
+		return <$thrift>.Response{}, err
+	}
+
+	<if .ReturnType>
+		success, resMeta, err := h.impl.<.Name>(ctx, reqMeta, <range .Arguments>args.<.Name>,<end>)
+	<else>
+		resMeta, err := h.impl.<.Name>(ctx, reqMeta, <range .Arguments>args.<.Name>,<end>)
+	<end>
+
+	hadError := err != nil
+	result, err := <$servicePackage>.<.Name>Helper.WrapResponse(<if .ReturnType>success,<end> err)
+
+	var response <$thrift>.Response
+	if err == nil {
+		response.IsApplicationError = hadError
+		response.Meta = resMeta
+		response.Body = result
+	}
+	return response, err
+}
+<end>
+`
+
+const clientTemplate = `
+// Code generated by thriftrw-plugin-yarpc
+// @generated
+
+<$pkgname := printf "%sclient" (lower .Service.Name)>
+package <$pkgname>
+
+<$yarpc     := import "github.com/yarpc/yarpc-go">
+<$transport := import "github.com/yarpc/yarpc-go/transport">
+<$thrift    := import "github.com/yarpc/yarpc-go/encoding/thrift">
+<$protocol  := import "github.com/thriftrw/thriftrw-go/protocol">
+<$context   := import "golang.org/x/net/context">
+
+// Interface is a client for the <.Service.Name> service.
+type Interface interface {
+	<if .Parent>
+		<$parentPath := printf "%s/yarpc/%sclient" .Parent.Package (lower .Parent.Name)>
+		<import $parentPath>.Interface
+	<end>
+
+	<range .Service.Functions>
+		<.Name>(
+			ctx <$context>.Context,
+			reqMeta <$yarpc>.CallReqMeta, <range .Arguments>
+				<.Name> <formatType .Type>,<end>
+		) <if .ReturnType> (<formatType .ReturnType>, <$yarpc>.CallResMeta, error)
+		<else> (<$yarpc>.CallResMeta, error)
+		<end>
+	<end>
+}
+
+</* TODO(abg): Pull the default routing name from a Thrift annotation? */>
+
+// New builds a new client for the <.Service.Name> service.
+//
+// 	client := <$pkgname>.New(dispatcher.Channel("<lower .Service.Name>"))
+func New(c <$transport>.Channel, opts ...<$thrift>.ClientOption) Interface {
+	return client{c: <$thrift>.New(<$thrift>.Config{
+		Service: "<.Service.Name>",
+		Channel: c,
+		Protocol: <$protocol>.Binary,
+	}, opts...)}
+}
+
+type client struct{ c <$thrift>.Client }
+
+<$service := .Service>
+<range .Service.Functions>
+
+<$servicePackage := import $service.Package>
+<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+
+func (c client) <.Name>(
+	ctx <$context>.Context,
+	reqMeta <$yarpc>.CallReqMeta, <range .Arguments>
+	_<.Name> <formatType .Type>,<end>
+) (<if .ReturnType>success <formatType .ReturnType>,<end> resMeta <$yarpc>.CallResMeta, err error) {
```

Comment:
Good point. That should get promoted to a response decode error since the result is a union. For a JSON endpoint with no validation, you can check expressly for body.success === null. It would be body.success === undefined otherwise.